packages/js/ai/src/hooks/useBackgroundRemoval.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+import { useState } from 'react';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import { createExtendedError } from '../utils/create-extended-error';
+
+export type BackgroundRemovalParams = {
+	imageFile: File;
+	returnedImageType?: 'png' | 'jpeg';
+	returnedImageSize?: 'hd' | 'medium' | 'preview';
+	token: string;
```

Comment:
I wonder if we should build in the token management here, so that the user doesn't have to provide one. Since they only last 2 minutes it can add some complexity, and we're doing so in the `useTextCompletion` hook. We have the `requestJetpackToken()` utility (currently [in the text-completion file](https://github.com/woocommerce/woocommerce/blob/45299a593e743509b90bfbb15c28190164072e03/packages/js/ai/src/utils/text-completion.ts#L32)) that could be moved to it's own file and reused. 

That said, we'll likely need to revisit all this when we want these features to work in environments that may not be able to get a token in this fashion 🤔.

Reply:
> I wonder if we should build in the token management here, so that the user doesn't have to provide one. 
> We have the requestJetpackToken() utility (currently [in the text-completion file](https://github.com/woocommerce/woocommerce/blob/45299a593e743509b90bfbb15c28190164072e03/packages/js/ai/src/utils/text-completion.ts#L32)) that could be moved to it's own file and reused.

I was thinking about this earlier as well, and came to the same conclusion; at the very least, the token generation should be moved out of `useTextCompletion()`, but, I'm also wondering about the repercussions of coupling token generation with `useBackgroundRemoval()`. For the reasons you said, I wonder if doing that would make it tricky for other consumers of this hook to get a token. 🤔

Reply:
I've addressed this in https://github.com/woocommerce/woocommerce/pull/40089/commits/f71ef861fa71066a6dcf3c1bef0ec70a5a58a22f